### PR TITLE
SWDEV-306204 - Update HIP doxygen and documents

### PR DESCRIPTION
**Pull request by jujiang-del**
Change-Id: I796eda5cb14dfd00d2a806c57c0550356fbbcdb2

### CODE REVIEW

**Review by gargrahul — DISMISSED** [review]

File: docs/markdown/hip_faq.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ HIP provides the following:
 - Memory management (hipMalloc(), hipMemcpy(), hipFree(), etc.)
 - Streams (hipStreamCreate(),hipStreamSynchronize(), hipStreamWaitEvent(),  etc.)
 - Events (hipEventRecord(), hipEventElapsedTime(), etc.)
-- Kernel launching (hipLaunchKernel is a standard C/C++ function that replaces <<< >>>)
+- Kernel launching (hipLaunchKernel/hipLaunchKernelGGL is the right API to call. hipLaunchKernelGGL is a standard C/C++ micro that can serve as an alternative way to launch kernek, and it can replace <<< >>>, if the user so desires).
```
Comment (Maetveis) [inline docs/markdown/hip_faq.md]:
```suggestion
- Kernel launching (hipLaunchKernel/hipLaunchKernelGGL is the preferred way of launching kernels. hipLaunchKernelGGL is a standard C/C++ macro that can serve as an alternative way to launch kernels, replacing the CUDA triple-chevron (`<<< >>>`) syntax).
```
typo fixes and small a tweak in wording

File: docs/markdown/hip_kernel_language.md [hunk 2]
```diff
@@ -708,12 +686,25 @@ The Cuda `__prof_trigger()` instruction is not supported.
 
 ## Assert
 
-The assert function is under development.
-HIP does support an "abort" call which will terminate the process execution from inside the kernel.
+The assert function is supported in HIP.
+There are two kinds of implementations for assert functions depending on the use sceneries,
+- One is only for the host version of assert, which is defined in assert.h
+- Another is the device version of assert, which is implemented in hip/amd_detail/hip_runtime.h.
+In HIP application, users should include the header file(s) accordingly while make assert function calls.
```
Comment (Maetveis) [inline docs/markdown/hip_kernel_language.md]: This is simply not true: hip/amd_detail/hip_runtime.h does not define `assert()` it defines `__assert_fail` which is called by the C library provided `assert` macro when the condition is false. It is valid to include `assert.h` in "device" code.

```cuda
#include <hip/hip_runtime.h>
#include <assert.h>

__global__ void f(int* ptr){
   assert(ptr);
   *ptr = 0;
}
```

File: docs/markdown/hip_programming_guide.md [hunk 3]
```diff
@@ -3,17 +3,19 @@
...
 
 ### Memory allocation flags
-hipHostMalloc always sets the hipHostMallocPortable and hipHostMallocMapped flags. Both usage models described above use the same allocation flags, and the difference is in how the surrounding code uses the host memory.
-
-hipHostMallocNumaUser is the flag to allow host memory allocation to follow numa policy set by user.
+There are flags parameter which can specify options how to allocate the memory, for example,
+hipHostMallocPortable, the memory is considered allocated by all contexts, not just the one on which the allocation is made.
+hipHostMallocMapped, will map the allocation into the address space for the current device, and the device pointer can be obtained with the API hipHostGetDevicePointer().
+hipHostMallocNumaUser is the flag to allow host memory allocation to follow numa policy by user.
+All allocation flags are independent, and can be used with combination without restriction, for instance, hipHostMalloc can be called with both hipHostMallocPortable and hipHostMallocMapped flags set. Both usage models described above use the same allocation flags, and the difference is in how the surrounding code uses the host memory.
```
Comment (Maetveis) [inline docs/markdown/hip_programming_guide.md]:
```suggestion
All allocation flags are independent, and can be used in any combination without restriction, for instance, hipHostMalloc can be called with both hipHostMallocPortable and hipHostMallocMapped flags set. Both usage models described above use the same allocation flags, and the difference is in how the surrounding code uses the host memory.
```

**Review by Maetveis — COMMENTED** [review]
Sorry for being nosy. I was just browsing around and found a few typos and inaccuracies.

File: docs/markdown/hip_kernel_language.md (hunk 2 above)
Comment (yxsamliu) [reply docs/markdown/hip_kernel_language.md]: You are right. the document should be updated as something like:

Users need to include assert.h to use assert. For assert to work in both device and host functions, users need to include "hip/hip_runtime.h".

File: docs/markdown/hip_faq.md (hunk 1 above)
Comment (jujiang-del) [reply docs/markdown/hip_faq.md]: Thanks for the review. Updated.

File: docs/markdown/hip_kernel_language.md (hunk 2 above)
Comment (jujiang-del) [reply docs/markdown/hip_kernel_language.md]: Thanks. Updated.

File: docs/markdown/hip_programming_guide.md (hunk 3 above)
Comment (jujiang-del) [reply docs/markdown/hip_programming_guide.md]: Thanks. Updated.

**Pull request merged**